src/middleware/prepareOutput.js

```diff
@@ -40,7 +40,7 @@ module.exports = function (options, excludedMap) {
         req.erm.result = options.filter ? options.filter.filterObject(req.erm.result, opts) : req.erm.result
       }
 
-      if (options.totalCountHeader && req.erm.totalCount !== 'undefined') {
+      if (options.totalCountHeader && typeof req.erm.totalCount === 'number') {
```

Comment:
The previous condition was a bit weird, I feel like this is much more robust for what we're trying to do here. `totalCount` should really be a number!